pkg/controller/sidecarterminator/sidecar_terminator_controller.go

```diff
@@ -0,0 +1,368 @@
+	err := wait.Poll(300*time.Microsecond, 3*time.Second, func() (done bool, err error) {
+		pod = &corev1.Pod{}
+		getErr := r.Client.Get(context.TODO(), types.NamespacedName{Name: originalPod.Name, Namespace: originalPod.Namespace}, pod)
+		if getErr != nil {
+			if errors.IsNotFound(getErr) {
+				return true, nil
+			}
+			return false, getErr
+		}
+
+		pod = updateImageForInPlaceUpdate(pod)
+		updateErr := r.Update(context.TODO(), pod)
+		if updateErr == nil {
+			return true, nil
+		}
+		return false, updateErr
+	})
+
+	if err != nil {
+		klog.V(3).Infof("SidecarTerminator -- Error occurred when inPlace update pod(%v/%v), error %v", originalPod.Namespace, originalPod.Name, err)
+	} else {
+		klog.V(3).Infof("SidecarTerminator -- InPlace update pod(%v/%v) successfully", originalPod.Namespace, originalPod.Name)
+	}
+
+	return reconcile.Result{}, err
+}
+
+func (r *ReconcileSidecarTerminator) execContainerStopStrategy(pod *corev1.Pod) (reconcile.Result, error) {
+	existingCRR := &appsv1alpha1.ContainerRecreateRequest{}
+	err := r.Get(context.TODO(), types.NamespacedName{Namespace: pod.Namespace, Name: getCRRName(pod)}, existingCRR)
+	if err == nil {
+		klog.V(3).Infof("SidecarTerminator -- CRR(%s/%s) has been existing, waiting for this CRR to complete", existingCRR.Namespace, existingCRR.Name)
+		return reconcile.Result{}, nil
+	} else if !errors.IsNotFound(err) {
+		klog.V(3).Infof("SidecarTerminator -- Error occurred when try to get CRR(%s/%s), error: %v", existingCRR.Namespace, existingCRR.Name, err)
+		return reconcile.Result{}, err
+	}
+
+	sidecars, _ := getSidecarNames(pod)
+	uncompletedSidecars := filterOutUncompletedSidecars(pod, sidecars)
+	var sidecarContainers []appsv1alpha1.ContainerRecreateRequestContainer
+	for _, name := range uncompletedSidecars.List() {
+		sidecarContainers = append(sidecarContainers, appsv1alpha1.ContainerRecreateRequestContainer{
+			Name: name,
+		})
+	}
+
+	crr := &appsv1alpha1.ContainerRecreateRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: pod.Namespace,
+			Name:      getCRRName(pod),
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(pod, pod.GroupVersionKind()),
+			},
+		},
+		Spec: appsv1alpha1.ContainerRecreateRequestSpec{
+			PodName:    getPodName(pod),
+			Containers: sidecarContainers,
+			Strategy: &appsv1alpha1.ContainerRecreateRequestStrategy{
+				FailurePolicy: appsv1alpha1.ContainerRecreateRequestFailurePolicyIgnore,
+			},
+		},
+	}
+
+	err = r.Create(context.TODO(), crr)
+	if err != nil {
+		klog.V(3).Infof("SidecarTerminator -- Error occurred when creating CRR(%v/%v), error %v", crr.Namespace, crr.Name, err)
+	} else {
+		klog.V(3).Infof("SidecarTerminator -- Creating CRR(%v/%v) successfully", crr.Namespace, crr.Name)
+	}
+
+	return reconcile.Result{}, err
+}
+
+func getCRRName(pod *corev1.Pod) string {
+	return fmt.Sprintf("%v-%v", getPodName(pod), pod.UID)
+}
+
+func getPodName(pod *corev1.Pod) string {
+	if pod.Name == "" {
+		return pod.GenerateName
```

Comment:
it seems impossible that a complete pod has empty "Name". and generateName of pods in the same workload(such as deployment) is the same, so name a CRR with generateName is not appropriate

Reply:
done